module/project/templates/helper.js

```diff
@@ -957,74 +958,122 @@ module.exports = class ProjectTemplatesHelper {
       * @name details
       * @param {String} templateId - Project template id.
       * @param {String} userId - logged in user id.
+      * @params {String} link - solution link.
       * @returns {Array} Project templates data.
      */
 
-    static details( templateId,userId ) {
+    static details( templateId="",link="",userId="" ) {
         return new Promise(async (resolve, reject) => {
             try {
-
+                let solutionsResult = {};
+                if(templateId == "" && link == ""){
+                    throw{
+                        status:HTTP_STATUS_CODE.bad_request.status,
+                        message:CONSTANTS.apiResponses.PROJECT_TEMPLATE_FETCH_ERROR
+                    }
+                }
                 let findQuery = {};
+                //get data when link is given
+                if(link){
+                    
+                    let queryData = {};
+                    queryData["link"] =link;
+
+                    let solutionDocument = 
+                    await coreService.solutionsDocuments(
+                        queryData,
+                        [
+                            "_id",
+                            "name",
+                            "programId",
+                            "programName",
+                            "externalId",
+                            "projectTemplateId",
+                            "link"
+                        ]
+                    );
 
-                let validateTemplateId = UTILS.isValidMongoId(templateId);
-
-                if( validateTemplateId ) {
-                  findQuery["_id"] = templateId;
-                } else {
-                  findQuery["externalId"] = templateId;
+                    if( !solutionDocument.success || solutionDocument.data.length === 0 ) {
+                        throw {
+                            message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                            status : HTTP_STATUS_CODE['bad_request'].status
+                        }
+                    }
+                    
+                    let solutiondata=solutionDocument.data;
```

Comment:
@VISHNUDAS-tunerlabs make a space before and after any assignment operator 
let solutiondata = solutionDocument.data;